fumadocs/components/example.tsx

```diff
@@ -1,108 +1,130 @@
-      {showCode && (
-        <div className={"z-10 flex flex-row gap-6 pb-4"}>
-          <a
-            className={
-              "nx-select-none nx-text-gray-600 hover:nx-text-black dark:nx-text-gray-200 dark:hover:nx-text-white flex flex-row items-center gap-1"
-            }
-            href={`${baseGitHubURL}${props.path}/`}
-            target="_blank"
-            rel="noreferrer">
-            <AiFillGithub />
-            <div className={"text-sm"}>GitHub</div>
-          </a>
-          <a
-            className={
-              "nx-select-none nx-text-gray-600 hover:nx-text-black dark:nx-text-gray-200 dark:hover:nx-text-white flex flex-row items-center gap-1"
-            }
-            href={`${baseStackBlitzURL}${props.path}/`}
-            target="_blank"
-            rel="noreferrer">
-            <SiStackblitz />
-            <div className={"text-sm"}>StackBlitz</div>
-          </a>
-        </div>
-      )}
-      <div className={"demo-contents h-96 overflow-auto rounded-lg"}>
-        <Component />
+    <div className="not-prose bg-fd-secondary border-fd-border flex h-[600px] flex-col rounded-xl border">
+      <ExampleDemoBar name={props.name} />
+      <div
+        className={"demo-contents bg-fd-background h-0 flex-1 rounded-xl p-4"}
+      >
+        <BlockNoteContext.Provider
+          value={{
+            colorSchemePreference: resolvedTheme === "dark" ? "dark" : "light",
+          }}
+        >
+          <Component />
+        </BlockNoteContext.Provider>
       </div>
-      {showCode ? (
-        props.children
-      ) : (
-        <div
-          className={
-            "relative flex h-96 flex-col items-center justify-center gap-2"
-          }>
-          <div className={"absolute h-1/2 w-1/2"}>
-            <div className={"cta-glow h-full w-full"}></div>
-          </div>
-          <div className={"z-10 flex w-2/3 flex-col items-center"}>
-            <SectionHeader>Pro Example</SectionHeader>
-            <p className={"text-center text-[#00000080] dark:text-[#FFFFFFB2]"}>
-              Get access to the full source code for pro examples by subscribing
-              to BlockNote Pro
-            </p>
-            <div className={"mt-8"}>
-              <CTAButton
-                href={"/pricing"}
-                color={"pro"}
-                size={"large"}
-                hoverGlow={true}>
-                Get BlockNote Pro
-              </CTAButton>
-            </div>
-            {/* {!props.isProExample?.userStatus && (
+    </div>
+  );
+}
+
+function ExampleCode(props: { name: string }) {
+  const [group, example] = props.name.split("/");
+  const tabs: Record<string, string> = (EXAMPLES_CODE_BLOCK_TABS as any)[group][
+    example
+  ];
+
+  return (
+    <Tabs items={Object.keys(tabs)}>
+      {Object.entries(tabs).map(([fileName, fileContent]) => (
+        <Tab key={fileName} value={fileName}>
+          <DynamicCodeBlock lang="tsx" code={fileContent} />
```

Comment:
This language should be derived from the file's extension, css syntax highlighting is weird now because of it